### ISSUE TITLE
Thoughts on question removeDuplicatesFromLinkedList

### ISSUE THREAD

**Issue by Joshmomel** [Joshmomel/algoexpert_ts#5]
# Task 🗒️ 
This task is to remove the duplicate node in the Linked list
![image](https://user-images.githubusercontent.com/50355313/115228059-a60b1100-a154-11eb-9736-a6d73f553763.png)

# How to approach? 🔢 
💰 In general, most tasks related to Linked list are about altering node.next.

In this task, we also need to come up with an approach to alter node.next, which ensures node.value NOT equal to node.next.vale

### Example 
1 -> 1 -> 1 -> 2 -> 3 =>  1 -> 3

So, it's not hard to think that we need an loop

```js
while(...) {
  if(!valueDuplicated) {
	...
  } else {
	`alter node.next`
  }
}
``` 

### The trick 
💬 **We can look ahead**

### But how?
Instead of checking `linkedlist` is null as the while loop condition like below
```js
  while (currentNode) {
	let nextNode = currentNode.next
    while(currentNode.value === nextNode.value) {
		nextNode = currentNode.next
  	}
	currentNode.next = nextNode
 	currentNode = nextNode
  }
```

We can check the next node like below
```js
  const list: Array<number> = [linkedList.value];
  let pointer = linkedList

  while (pointer.next) {
    const v = pointer.next.value;
	...
    pointer.next = pointer.next.next

  }
```

# Summary
In summary, by looking ahead, pointer only move to the right place instead moving one by one. However, both approach have same complexity

**Comment by Joshmomel** [comment]
Checkout the [Solution](https://github.com/Joshmomel/algoexpert_ts/blob/master/src/easy/removeDuplicatesFromLinkedList.ts)

**Comment by Joshmomel** [comment]
Checkout the [Solution](https://github.com/Joshmomel/algoexpert_ts/blob/master/src/easy/removeDuplicatesFromLinkedList.ts)
